File: python/Singleton/complete/SingletonEager.py
class SingletonEager:
    obj = None

    #initialization of static variable to owner class not allowed in python
    #so we do it in constructor
    def __init__(self):
        self.log = ""
        if(SingletonEager.obj == None):
            SingletonEager.obj = 1 #placeholder
            SingletonEager.obj = SingletonEager()

    @staticmethod
    def get_instance():
        return SingletonEager.obj

    def add_to_log(self, string):
        if(self.log == ""):
            self.log += string
        else:
            self.log += "\n"+string

    def get_log(self):
        return self.log

File: java/Singleton/complete/SingletonEager.java

public class SingletonEager {
  public static SingletonEager obj = new SingletonEager();
  private String log;

  private SingletonEager(){
    //set the log to blank
    log = "";
  }

  public static SingletonEager get_instance(){
    //return the Singleton object
    return obj;
  }

  public void add_to_log(String s){
    //add String s to the current log
    if(log.equals(""))
      log = log + s;
    else
      log = log + "\n" + s;
  }

  public String get_log(){
    return log;
  }
}
